llvm/lib/Target/AIE/AIEBaseHardwareLoops.cpp

```diff
@@ -206,48 +279,73 @@ bool AIEBaseHardwareLoops::ProcessLoop(MachineLoop *ML) {
     }
   }
 
-  Expand(LoLoop);
+  expand(LoLoop);
   return true;
 }
 
-void AIEBaseHardwareLoops::ExpandLoopStart(LowOverheadLoop &LoLoop) {
+void AIEBaseHardwareLoops::expandLoopStart(LowOverheadLoop &LoLoop) {
   LLVM_DEBUG(dbgs() << "AIE Loops: Expanding LoopStart.\n");
 
   if (LoLoop.isJNZDLoop()) {
     LLVM_DEBUG(dbgs() << "AIE Loops: JNZD loop. Nothing to be done.\n");
     return;
   }
 
-  // Future: Handle true hardware loop setup code
-  llvm_unreachable("Implement true hardware loop support");
+  if (LoLoop.isZeroOverheadLoop()) {
```

Comment:
Do we really need to test this condition? Is there any situation where this if will evaluate to false?

Reply:
I'll check. Modify it to an assert if it can be.